lib/octopus_serverspec_extensions/type/octopus_deploy_worker.rb

```diff
@@ -0,0 +1,173 @@
+require 'serverspec'
+require 'serverspec/type/base'
+require 'net/http'
+require 'json'
+
+module Serverspec::Type
+  class OctopusDeployWorker < Base
+    @worker = nil
+    @serverUrl = nil
+    @apiKey = nil
+    @serverSupportsSpaces = nil
+    @spaceId = nil
+    @spaceFragment = ""
+
+    def initialize(serverUrl, apiKey, instance, spaceId = 'Spaces-1')
+      @name = "Octopus Deploy Worker #{instance}"
+      @runner = Specinfra::Runner
+      @serverUrl = serverUrl
+      @apiKey = apiKey
+      @spaceId = spaceId
+
+      if (serverUrl.nil?)
+        raise "'serverUrl' was not provided. Unable to connect to Octopus server to validate configuration."
+      end
+      if (apiKey.nil?)
+        raise "'apiKey' was not provided. Unable to connect to Octopus server to validate configuration."
+      end
+
+      if (exists?)
+        thumbprint = `"c:\\program files\\Octopus Deploy\\Tentacle\\Tentacle.exe" show-thumbprint --console --nologo --instance #{instance}`
```

Comment:
Should we use the env var for program files?
Should we look this up in the registry?
Should we just ignore it until it becomes a problem?

Reply:
I'm in favour of Option 3 :trollface: